test/images/agnhost/fakeregistryserver/registryserver.go

```diff
@@ -0,0 +1,122 @@
+/*
+Copyright 2025 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package fakeregistryserver
+
+import (
+	"fmt"
+	"log"
+	"net/http"
+	"os"
+	"strings"
+
+	"github.com/spf13/cobra"
+)
+
+var (
+	port        int
+	host        string
+	registryDir string
+	private     bool
+)
+
+func init() {
+	CmdFakeRegistryServer.Flags().IntVar(&port, "port", 5000, "Port number.")
+	CmdFakeRegistryServer.Flags().StringVar(&host, "host", "0.0.0.0", "Host address.")
+	CmdFakeRegistryServer.Flags().StringVar(&registryDir, "registry-dir", "/registry", "Directory containing the registry data.")
+	CmdFakeRegistryServer.Flags().BoolVar(&private, "private", false, "Enable basic authentication with static credentials (test:test)")
+}
+
+// CmdFakeRegistryServer is the cobra command for the fake registry server
+var CmdFakeRegistryServer = &cobra.Command{
+	Use:   "fake-registry-server",
+	Short: "Starts a fake registry server for testing",
+	Long:  `Starts a fake registry server that serves static OCI image files`,
+	Run:   main,
+}
+
+func main(cmd *cobra.Command, args []string) {
+	mux := http.NewServeMux()
+	mux.HandleFunc("/v2/", handleV2)
+
+	var handler http.Handler = mux
+	if private {
+		handler = auth(mux)
+	}
+
+	serverAdr := fmt.Sprintf("%s:%d", host, port)
```

Comment:
```suggestion
	serverAdr := fmt.Sprintf(":%d", port)
```

Reply:
That means host become a static var, or constant?

Reply:
Yes, I don't think we need it configurable, do we? Not even for tests.

Reply:
Done 👍